homeassistant/components/vulcan/config_flow.py

```diff
@@ -0,0 +1,342 @@
+        for entry in self.hass.config_entries.async_entries(DOMAIN):
+            credentials_list[entry.entry_id] = entry.data["account"]["UserName"]
+
+        if user_input is not None:
+            entry = self.hass.config_entries.async_get_entry(user_input["credentials"])
+            keystore = Keystore.load(entry.data["keystore"])
+            account = Account.load(entry.data["account"])
+            client = Vulcan(keystore, account)
+            try:
+                students = await client.get_students()
+            except VulcanAPIException as err:
+                if str(err) == "The certificate is not authorized.":
+                    return await self.async_step_auth(
+                        errors={"base": "expired_credentials"}
+                    )
+                _LOGGER.error(err)
+                return await self.async_step_auth(errors={"base": "unknown"})
+            except ClientConnectionError as err:
+                _LOGGER.error("Connection error: %s", err)
+                return await self.async_step_select_saved_credentials(
+                    errors={"base": "cannot_connect"}
+                )
+            except Exception:  # pylint: disable=broad-except
+                _LOGGER.exception("Unexpected exception")
+                return await self.async_step_auth(errors={"base": "unknown"})
+            finally:
+                await client.close()
+            if len(students) == 1:
+                student = students[0]
+                await self.async_set_unique_id(str(student.pupil.id))
+                self._abort_if_unique_id_configured()
+                return self.async_create_entry(
+                    title=f"{student.pupil.first_name} {student.pupil.last_name}",
+                    data={
+                        "student_id": str(student.pupil.id),
+                        "keystore": keystore.as_dict,
+                        "account": account.as_dict,
+                    },
+                )
+            # pylint:disable=attribute-defined-outside-init
+            self.account = account
+            self.keystore = keystore
+            self.students = students
+            return await self.async_step_select_student()
+
+        data_schema = {
+            vol.Required(
+                "credentials",
+            ): vol.In(credentials_list),
+        }
+        return self.async_show_form(
+            step_id="select_saved_credentials",
+            data_schema=vol.Schema(data_schema),
+            errors=errors,
+        )
+
+    async def async_step_add_next_config_entry(self, user_input=None):
+        """Flow initialized when user is adding next entry of that integration."""
+        existing_entries = []
+        for entry in self.hass.config_entries.async_entries(DOMAIN):
+            existing_entries.append(entry)
+
+        errors = {}
+        if user_input is not None:
+            if user_input["use_saved_credentials"]:
+                if len(existing_entries) == 1:
```

Comment:
As above. We can invert and go to the async_step_select_saved_credentials if true.